be1-go/hub/standard_hub/hub_state/ThreadSafeSlice.go

```diff
@@ -0,0 +1,29 @@
+package hub_state
+
+import (
+	"golang.org/x/exp/slices"
+	"sync"
+)
+
+type ThreadSafeSlice[E comparable] struct {
+	sync.RWMutex
+	els []E
+}
+
+func NewThreadSafeSlice[E comparable]() ThreadSafeSlice[E] {
+	return ThreadSafeSlice[E]{
+		els: make([]E, 0),
+	}
+}
+
+func (i *ThreadSafeSlice[E]) Append(elems ...E) {
+	i.Lock()
+	defer i.Unlock()
+	i.els = append(i.els, elems...)
+}
+
+func (i *ThreadSafeSlice[E]) Contains(elem E) bool {
+	i.RLock()
+	defer i.RUnlock()
+	return slices.Contains(i.els, elem)
+}
```

Comment:
Shouldn't you add a method to remove an element from the blacklist in case we need it later?

Reply:
I think we should keep the code as simple as possible and add functions only when we need them